Documentation/operations/metrics.rst

```diff
@@ -207,6 +207,18 @@ Name                                       Labels
 ``services_events_total``                                                                     Number of services events labeled by action type
 ========================================== ================================================== ========================================================
 
+Cluster health
+~~~~~~~~~~~~~~
+
+========================================== ================================================== ========================================================
+Name                                       Labels                                             Description
+========================================== ================================================== ========================================================
+``unreachable_nodes``                                                                         Number of nodes that cannot be reached
+``unreachable_health_endpoints``                                                              Number of health endpoints that cannot be reached
+``ip_addresses``                                                                              Number of allocated IP addresses
```

Comment:
That metric seems to have a label.

Reply:
> @pchaigno @joestringer Can you please have a look and tell me if the metrics are listed under correct headings , and how the labels are assigned to corresponding metrices ?

Sorry but i asked this earlier also . How to find the labels of the metrics ?

Reply:
```
$ git grep "func NewDesc(" vendor/github.com/prometheus/
vendor/github.com/prometheus/client_golang/prometheus/desc.go:func NewDesc(fqName, help string, variableLabels []string, constLabels Labels) *Desc {
```
I'm not sure if `variableLabels` and `constLabels` both need to be documented but checking existing entries should answer that question. Then:
```
$ git grep -C3 ip_addresses pkg/metrics/
pkg/metrics/status.go-                  nil, nil,
pkg/metrics/status.go-          ),
pkg/metrics/status.go-          ipAddressesDesc: prometheus.NewDesc(
pkg/metrics/status.go:                  prometheus.BuildFQName(Namespace, "", "ip_addresses"),
pkg/metrics/status.go-                  "Number of allocated IP addresses",
pkg/metrics/status.go-                  []string{"family"}, nil,
pkg/metrics/status.go-          ),
$ git grep -C3 unreachable_nodes pkg/metrics/
pkg/metrics/status.go-                  []string{"family"}, nil,
pkg/metrics/status.go-          ),
pkg/metrics/status.go-          unreachableNodesDesc: prometheus.NewDesc(
pkg/metrics/status.go:                  prometheus.BuildFQName(Namespace, "", "unreachable_nodes"),
pkg/metrics/status.go-                  "Number of nodes that cannot be reached",
pkg/metrics/status.go-                  nil, nil,
pkg/metrics/status.go-          ),
$ git grep -C3 unreachable_health_endpoints pkg/metrics/
pkg/metrics/status.go-                  nil, nil,
pkg/metrics/status.go-          ),
pkg/metrics/status.go-          unreachableHealthEndpointsDesc: prometheus.NewDesc(
pkg/metrics/status.go:                  prometheus.BuildFQName(Namespace, "", "unreachable_health_endpoints"),
pkg/metrics/status.go-                  "Number of health endpoints that cannot be reached",
pkg/metrics/status.go-                  nil, nil,
pkg/metrics/status.go-          ),
$ git grep -C3 controllers_failing pkg/metrics/
pkg/metrics/status.go-          ciliumClient: ciliumClient,
pkg/metrics/status.go-          healthClient: healthClient,
pkg/metrics/status.go-          controllersFailingDesc: prometheus.NewDesc(
pkg/metrics/status.go:                  prometheus.BuildFQName(Namespace, "", "controllers_failing"),
pkg/metrics/status.go-                  "Number of failing controllers",
pkg/metrics/status.go-                  nil, nil,
pkg/metrics/status.go-          ),
```